src/main/java/seedu/address/logic/parser/profile/ProfileCommandParser.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic.parser.profile;
+
+import static seedu.address.commons.core.Messages.MESSAGE_FLAG_NOT_SPECIFIED;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.logic.commands.profile.AddProfileCommand;
+import seedu.address.logic.commands.profile.EditProfileCommand;
+import seedu.address.logic.commands.profile.ProfileCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ProfileCommand object
+ */
+public class ProfileCommandParser implements Parser<ProfileCommand> {
+    private static final Pattern PROFILE_COMMAND_FORMAT = Pattern.compile(
+            "\\s+(-(?<profileOption>\\S*)(\\s+(?!-)\\S+)?)(\\s+(?!-)\\S*.)*");
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ProfileCommand
+     * and returns an ProfileCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ProfileCommand parse(String args) throws ParseException {
+        if (!args.matches("\\s+-.*")) {
+            Matcher matcher = Pattern.compile("(\\s+-\\S*)").matcher(args);
+            long numFlags = matcher.results().count();
+            if (numFlags == 0) {
+                throw new ParseException(String.format(MESSAGE_FLAG_NOT_SPECIFIED, ProfileCommand.PROFILE_FORMAT));
+            } else if (numFlags == 1) {
+                throw new ParseException(ProfileCommand.OPTION_WRONG_ORDER);
+            } else {
+                throw new ParseException(ProfileCommand.OPTION_WRONG_ORDER_NO_MULTIPLE);
+            }
```

Comment:
I really like how you dealt with all the different scenarios when the flag was given but not in the correct format.